apps/nfid-frontend-e2e/src/steps/swap.ts

```diff
@@ -0,0 +1,65 @@
+import { When } from "@cucumber/cucumber"
+import Assets from "../pages/assets.js"
+import Activity from "../pages/activity.js"
+
+let currentSourceTokenBalance: number
+let currentTargetTokenBalance: number
+let expectedTargetTokenAmount: number
+let sourceTokenAmountToSwap: number
+When(/^Verifying that the swap transaction is success$/, async () => {
+  await Assets.SwapDialog.successTitle.waitForDisplayed()
+  await Assets.SwapDialog.closeButton.click()
+})
+
+When(/^User clicks the Swap tokens button$/, async () => {
+  await Assets.SwapDialog.swapTokensButton.click()
+  await browser.waitUntil(async () => {
+    return await Assets.SwapDialog.successTitle.getText() == "Swap successful"
+  }, { timeout: 60000, timeoutMsg: "Swap successful title didn't appear" })
+})
+
+When(/^User sets the (.*) token to (.*)$/, async (tokenRole: string, token: string) => {
+  await Assets.SwapDialog.getChooseTokenModalButton(tokenRole).then(async (it) => {
+    await it.waitForClickable()
+    await it.click()
+  })
+  await Assets.SwapDialog.getSearchTokenInputField(tokenRole).then(async (it) => {
+    await it.waitForClickable()
+    await it.setValue(token)
+  })
+  await Assets.SwapDialog.getTokenByNameFromList(tokenRole, token).then(async (it) => {
+    await it.waitForDisplayed({ timeout: 70000 })
+    await it.click()
+  })
+})
+
+When(/^User sets amount to swap to (.*)$/, async (amount: string) => {
+  await Assets.SwapDialog.getTokenAmountField("source").setValue(amount)
+  await browser.waitUntil(async () => {
+    return await Assets.SwapDialog.getTokenAmountField("target").isDisplayed()
+      && await (await Assets.SwapDialog.getTokenAmountField("target")).getAttribute("value") != 0.00
```

Comment:
double await

Reply:
and may be just parseFloat(await Assets.SwapDialog.getTokenAmountField("target")).getAttribute("value")), not sure value is number not string, because in the next lines you use parseFloat from the same attribute